[Terence Chong] iP

cxyterence opened the pull request:
**GustavoBot**
> "I like to think I see things in people." - Gustavo Fring (Breaking Bad)

_This project has nothing to do with the plot of hit series Breaking Bad,_ however it:

- serves you with impeccable service
- speaks in full sentences
- keeps tracks of things just for you
- most importantly, it **respects** you

If you would like to join the family, please:
1. download it from [here](https://github.com/cxyterence/ip/releases/tag/v0.1)
2. launch it
3. ~~look at~~ chat with GustavoBot 💪🏾🤓

Features:

- [ ] Manages tasks for you
- [ ] Manages deadlines for you
- [ ] Manages reminders for you (coming soon) 

If you're a Java programmer, GustavoBot allows you to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

cxyterence committed:
changed DUKE to GustavoBot

cxyterence committed:
added ability to store and display whatever text entered by the user, when requested

cxyterence committed:
added ability to mark task as done

cxyterence committed:
added support for tracking three types of tasks

cxyterence committed:
added support for error handling due to incorrect user inputs

cxyterence committed:
added support for deleting tasks from list

cxyterence committed:
use the input/output redirection technique to semi-automate the testing of my version of Duke.

cxyterence committed:
add ability to save tasks in the hard disk automatically whenever the task list changes. add the ability to load data from the hard disk when Duke starts up.

cxyterence committed:
add ability to store deadlines as java.time.LocalDate in task objects, and print the deadlines in a different format.

cxyterence committed:
Merge branch 'branch-Level-8'

cxyterence committed:
removed redundancies.

cxyterence committed:
Refactor the code to extract out closely related code as classes.

cxyterence committed:
Organise the classes into suitable java packages.

cxyterence committed:
set up Gradle to automate some of the build tasks of the project.

cxyterence committed:
fix bugs

cxyterence committed:
Revert "fix bugs"

This reverts commit 22ef0bd925df26f41cb2708338498c0f14e30e39.

cxyterence committed:
Add JUnit tests to test the behavior of the code.

cxyterence committed:
Package the app as an executable JAR file that can be distributed easily.

cxyterence committed:
Add JavaDoc comments to the code.

cxyterence committed:
Tweak the code to comply with a coding standard.

cxyterence committed:
Give users a way to find a task by searching for a keyword.

shwene commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,36 @@
+package duke;
+
+import duke.command.*;
```
I think wildcard imports are not allowed, perhaps you could place Parser.java in duke.command?

shwene commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,122 @@
+package duke;
+
+import duke.exception.EmptyDescException;
+import duke.task.*;
```
Same import issue, though more feasible to state all the task classes here

shwene commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,34 @@
+    public void showWelcome() {
+        String intro = "Hello! My name is GustavoBot, but you can call me Gus\n"
+                + "How may I help you today?";
+
```
Excess line could be extracted out

shwene commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,44 @@
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+
```
Could be an excess line here as there is no line break between duke.Ui and duke.task import line in the other Command classes

shwene commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,37 @@
+
+public class EventCommand extends Command {
+    private String desc;
+    private String at;
```
Perhaps this could be `location` / `atTiming` to be more descriptive

shwene commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,22 @@
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        ListIterator<Task> iterate = tasks.getIterator();
+        ui.printMessage("Here are the tasks you currently have:");
+        int qty = 0;
```
Perhaps this could be `quantity`

shwene commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+
+public class Deadline extends Task {
+    protected String by;
+    protected LocalDate d;
```
Perhaps a more descriptive variable name could be used

shwene reviewed (commented):
Looks neat overall, just a couple of minor minor issues!

cxyterence committed:
Add a GUI to Duke. Use the JavaFX technology to implement the GUI.

cxyterence committed:
Add Java assert feature

Integer variables used as counters, in some classes that inherit from
Command, cannot be less than zero.
In some classes that inherit from Command, the task description
and/or the date cannot be empty.

So let's assume that these are not the case in the code.
If the cases which they do happen, we let the runtime detect an
assertion failure to inform us that there are potentially bugs that
cause it to happen.

cxyterence committed:
Examine and refactor the code to improve the code quality

Some snippets of the code are hard to understand due to factors like
naming and not following coding standards as defined by the module.

Let's examine and refactor codes that are:
- difficult to read
- named poorly to be understood
- not complying with standards

cxyterence committed:
Modify the data file that Duke/GustavoBot loads at the start

cxyterence committed:
Merge pull request #1 from cxyterence/branch-A-Assertions

Add Java assert feature

cxyterence committed:
Merge branch 'master' into branch-A-CodeQuality

cxyterence committed:
Modified data save file which Duke/GustavoBot loads at the start

cxyterence committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:
  Examine and refactor the code to improve the code quality

cxyterence committed:
Add ability to tag tasks with a description

It is good to allow users to have a way to tag their tasks for different
purposes.

Let's create a new class TagCommand that inherits from Command, and it
serves as the command that will set up Duke/GustavoBot to tag a task.

cxyterence committed:
Modified data save file which Duke/GustavoBot loads at the start

cxyterence committed:
Merge branch 'branch-BCD-Extension'

* branch-BCD-Extension:
  Add ability to tag tasks with a description

cxyterence committed:
Give a unique personality.

cxyterence committed:
Improve the GUI to make it more polished.

cxyterence committed:
Add a User Guide to the project.

cxyterence committed:
Fix bugs related to build.gradle.

cxyterence committed:
Resolve bugs related to READEME.md.

cxyterence committed:
Fix bugs related to README.md.

cxyterence committed:
Fix bugs related to README.md
